lib/blockchain.go

```diff
@@ -2633,15 +2640,172 @@ func (bc *Blockchain) processBlockPoW(desoBlock *MsgDeSoBlock, verifySignatures
 }
 
 // processBlockPoS runs the Fast-Hotstuff block connect and commit rule as follows:
-// 1. Validate on an incoming block and its header
-// 2. Store the block in the db
-// 3. Resolves forks within the last two blocks
-// 4. Connect the block to the blockchain's tip
-// 5. If applicable, flush the incoming block's grandparent to the DB
-// 6. Notify the block proposer, pacemaker, and voting logic that the incoming block has been accepted
-func (bc *Blockchain) processBlockPoS(desoBlock *MsgDeSoBlock, verifySignatures bool) (_isMainChain bool, _isOrphan bool, err error) {
+//  1. Validate on an incoming block, its header, its block height, the leader, and its QCs (vote or timeout)
+//  2. Store the block in the block index
+//  3. Determine if we're missing a parent block of this block and any of its parents from the block index.
+//     If so, return the hash of the missing block.
+//  4. Resolves forks within the last two blocks
+//  5. Connect the block to the blockchain's tip
+//  6. Update in-memory struct holding uncommitted block. @sofonias - should this happen AFTER we run the commit rule?
+//  7. Run the commit rule - If applicable, flushes the incoming block's grandparent to the DB
+//  8. Update the currentView to this new block's view + 1
+//  9. Notify the block proposer, pacemaker, and voting logic that the incoming block has been accepted by returning success.
```

Comment:
No need to have ProcessBlockPoS do this. The server._handleBlock function would pass the status to the Fast-HotStuff event loop
```suggestion
```